group-coordinator/src/test/java/org/apache/kafka/coordinator/group/GroupMetadataManagerTest.java

```diff
@@ -9863,8 +9863,8 @@ public void testConsumerGroupHeartbeatWithPreparingRebalanceClassicGroup() throw
 
             // Newly joining member 3 bumps the group epoch. A new target assignment is computed.
             CoordinatorRecordHelpers.newGroupEpochRecord(groupId, 1),
-            CoordinatorRecordHelpers.newTargetAssignmentRecord(groupId, memberId1, assignor.targetPartitions(memberId1)),
             CoordinatorRecordHelpers.newTargetAssignmentRecord(groupId, memberId3, assignor.targetPartitions(memberId3)),
+            CoordinatorRecordHelpers.newTargetAssignmentRecord(groupId, memberId1, assignor.targetPartitions(memberId1)),
```

Comment:
Why do we need this change?

Reply:
The tests are failing with the previous order

Reply:
Why did it change? My understanding is that changes made should not impact the assignor’s logic.

Reply:
I'm not sure tbh

Reply:
It would be great if we could better understand this.

Reply:
I wonder if it could be caused by this change: https://github.com/apache/kafka/pull/16068/files#diff-a193e75a338ac43ed80fdd0ec98858282cea2bcba8ba5d9da4d22e7f52e8853aR296. If it is, let's revert it for now.

Reply:
Oh yes it's exactly that! Thank you so much!